### PR TITLE
Set 0.5.1 release with bumped dependencies

### PR DESCRIPTION
**Pull request by alvarobartt**
## Description

This PR sets the 0.5.1 version for the `huggingface-inference-toolkit` and bumps some of the dependencies as of the recent releases, including the ping of the `huggingface_hub` that deprecates the `cached_download` affecting `diffusers<0.31.0`.

Some of the version bumps are described below:

- `transformers` 4.45.2 includes Llama 3.2, MLlama, and much more (see release notes at https://github.com/huggingface/transformers/releases/tag/v4.45.0)
- `diffusers` 0.31.0 includes Stable Diffusion 3.5 and much more (see release notes at https://github.com/huggingface/diffusers/releases/tag/v0.31.0)
- ~`huggingface_hub` 0.26.1 bumped to avoid issues with deprecated `cached_download` affecting older versions of `diffusers` i.e. 0.30.0 or below, so it was a sensible dependency~ Instead it has been pined to <0.26 to prevent issues if an older version of `diffusers` is installed within the custom requirements, but it could be as well pined to a greater version (both tested with latest diffusers and both work great, the only issue is with older diffusers)
- Also bumped `torch` 2.3.1, `peft` 0.13.2, `accelerate` 1.0.1 stable release, and `sentence-transformers` 3.2.1

Additionally, this PR also extends support for newer models such as Llama 3.2, MLlama, Stable Diffusion 3.5, and some more; due to the dependency bump.

### CODE REVIEW

**Review by philschmid — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -5,16 +5,16 @@
...
 # Ubuntu packages
 # libsndfile1-dev: torchaudio requires the development version of the libsndfile package which can be installed via a system package manager. On Ubuntu it can be installed as follows: apt install libsndfile1-dev
 # ffmpeg: ffmpeg is required for audio processing. On Ubuntu it can be installed as follows: apt install ffmpeg
 # libavcodec-extra : libavcodec-extra  includes additional codecs for ffmpeg
 
 install_requires = [
-    "transformers[sklearn,sentencepiece,audio,vision,sentencepiece]==4.44.0",
-    "huggingface_hub[hf_transfer]==0.24.5",
+    "transformers[sklearn,sentencepiece,audio,vision,sentencepiece]==4.46.0",
+    "huggingface_hub[hf_transfer]<0.26",
```
Comment (philschmid) [inline setup.py:17]: du we need this since `diffusers>0.29.0`?

**Review by philschmid — COMMENTED** [review]
one comment

File: setup.py (hunk 1 above)
Comment (philschmid) [reply setup.py:17]: The container will be used on new deployments. I think its fine if they break as it seems the decision was to rather help them upgrade their diffusers version etc.

Comment (alvarobartt) [reply setup.py:17]: Great then reverting that and updating the PR description before merging, thanks!

**Pull request merged**